qiita_pet/static/js/sampleTemplateVue.js

```diff
@@ -160,17 +160,45 @@ Vue.component('sample-template-page', {
      **/
     updateSampleTemplate: function() {
       let vm = this;
-      $.ajax({
-        url: vm.portal + '/study/description/sample_template/',
-        type: 'PATCH',
-        data: {'op': 'replace', 'path': vm.studyId + '/data/', 'value': $('#file-select').val()},
-        success: function(data) {
-          vm.startJobCheckInterval(data['job']);
-        },
-        error: function (object, status, error_msg) {
-          bootstrapAlert("Error updating sample template: " + error_msg, "danger")
-        }
-      });
+      var file = $('#st-direct-upload')[0].files[0];
+
+      // We can have 2 scenarios: direct file upload or path from upload
+      // folder. The former is sent via PATCH using defauls, the latter
+      // via POST using "processData: false, contentType: false"
```

Comment:
Why would the contentType be a boolean value? It seems non-standard/unexpected.

Reply:
good question and to be honest not sure, from: https://stackoverflow.com/a/5976031; adding the ref to the code